pkg/controllers/cassandra/nodepool/nodepool.go

```diff
@@ -68,6 +79,61 @@ func (e *defaultCassandraClusterNodepoolControl) removeUnusedStatefulSets(
 	return nil
 }
 
+// Return the lowest int from a slice of ints
+func min(v []int) (m int) {
+	if len(v) > 0 {
+		m = v[0]
+	} else {
+		panic("min called with empty slice")
+	}
+	for _, e := range v {
+		if e < m {
+			m = e
+		}
+	}
+	return
+}
+
+func (e *defaultCassandraClusterNodepoolControl) labelSeedNodes(
+	cluster *v1alpha1.CassandraCluster,
+	set *appsv1beta1.StatefulSet,
+) error {
+	var nodepoolPods []*v1.Pod
+	allPods, err := e.pods.Pods(cluster.Namespace).List(labels.Everything())
+	if err != nil {
+		return err
+	}
+	for _, pod := range allPods {
+		if metav1.IsControlledBy(pod, set) {
+			nodepoolPods = append(nodepoolPods, pod)
+		}
+	}
+
+	// if there are no pods, return early
+	if len(nodepoolPods) < 1 {
+		return nil
+	}
+
+	// Choose the lowest StatefulSet member and mark it as a seed
+	numberedPods := make(map[int]*v1.Pod)
+	ns := []int{}
+	for _, p := range nodepoolPods {
+		elements := strings.Split(p.Name, "-")
+		n, err := strconv.Atoi(elements[len(elements)-1])
+		if err != nil {
+			return err
+		}
+		numberedPods[n] = p
+		ns = append(ns, n)
+	}
```

Comment:
By selecting the lowest number pod, we'll be changing which pod is used as a seed every time the 0th pod fails.

Instead, we should *always* use `fmt.Sprintf("%s-0", p.Name)`.

Reply:
:+1: doing it this way makes it much simpler, I've switched the latest version to this